pkg/rc/replication_controller.go

```diff
@@ -316,6 +318,70 @@ func (rc *replicationController) addPods(current types.PodLocations, eligible []
 	return nil
 }
 
+func (rc *replicationController) allocateNodes(currentNodes []types.NodeName, eligibleNodes []types.NodeName) error {
+	var err error
+	if rc.AllocationStrategy == fields.CattleStrategy {
+		err = rc.allocateCattleNodes(currentNodes, eligibleNodes)
+	} else {
+		err = rc.allocatePetNodes(currentNodes, eligibleNodes)
+	}
+
+	if err != nil {
+		return util.Errorf("Unable to allocate nodes: %s", err)
+	}
+
+	return nil
+}
+
+func (rc *replicationController) allocateCattleNodes(currentNodes []types.NodeName, eligibleNodes []types.NodeName) error {
+	rcStatus, _, err := rc.rcStatusStore.Get(rc.ID())
+	isNoNodeTransferInProgress := (err != nil && statusstore.IsNoStatus(err)) ||
+		(err == nil && rcStatus.NodeTransfer == nil)
+	if isNoNodeTransferInProgress {
+		rc.mu.Lock()
+		newNodes, err := rc.scheduler.AllocateNodes(rc.Manifest, rc.NodeSelector, len(currentNodes)+1)
```

Comment:
I think you should just pass `1` here rather than `len(currentNodes)+1`. We could implement the server to know what you mean but that will require more calls between p2-resource-api and hoist and also the RC loop here has access to the full list of nodes, so it won't do a double allocate

Reply:
Sounds good, I just misunderstood the intention of that parameter. I remember debating between a couple of interpretations. I should've just asked 🙃

Reply:
yeah at some point I thought we needed the total, then realized we didn't! sorry for the churn